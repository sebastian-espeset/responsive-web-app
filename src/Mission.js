import React from 'react'
import { StyledMissionBody, StyledMissionContainer,
          StyledMissionHeadline,
          StyledToothBrushImageContainer,
          StyledToothGlassesImageContainer
      } from './styles/Mission.style'
import brush from './assets/brush.png';
import glasses from './assets/glasses.png';

export default function Mission() {
  return (
    <StyledMissionContainer>
      <StyledToothBrushImageContainer>
        <img src={brush} alt="toothbrush"/>
      </StyledToothBrushImageContainer>
      <StyledMissionHeadline data-testid = 'mission-headline-test-id'>
        Our Mission & Vision
      </StyledMissionHeadline>
      <div>────────</div>
      <StyledMissionBody>
      Beam Dental was founded on the idea that everyone deserves access to dental care, centering our mission around closing the gap of the 100 million Americans who don't. By blending innovative technology with traditional insurance policies, we've been able to bring incredible value to the employee benefits market—both disrupting insurance forever and delivering on our mission.<br></br><br></br> Beam’s vision is to deliver a fundamentally unique approach to coverage by incorporating dental hygiene behavior into policy pricing at renewal.
      </StyledMissionBody>
      <StyledToothGlassesImageContainer>  
      <img src={glasses} alt="eye glasses"/>
      </StyledToothGlassesImageContainer>
    </StyledMissionContainer>
  )
}
