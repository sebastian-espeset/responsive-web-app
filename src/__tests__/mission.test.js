import {render, screen, cleanup} from '@testing-library/react';
import renderer from 'react-test-renderer';
import Mission from '../Mission';

afterEach(()=>{
    cleanup();
});

test('Missions section headline should render',()=>{
    //Arrange
    render(<Mission/>)
    //Act
    const missionHeadline = screen.getByTestId('mission-headline-test-id');
    //Assert
    expect(missionHeadline).toBeInTheDocument();
});

// test will create a 'snapshot' of the html file, and then compare it to potential changes made    
test('mission component to match snapshot',()=>{
    const tree = renderer.create(<Mission/>).toJSON();
    expect(tree).toMatchSnapshot();
})