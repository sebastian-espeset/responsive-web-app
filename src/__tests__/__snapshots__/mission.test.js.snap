// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`mission component to match snapshot 1`] = `
<div
  className="sc-bczRLJ VBQCD"
>
  <div
    className="sc-dkzDqf fmcnBM"
  >
    <img
      alt="toothbrush"
      src="brush.png"
    />
  </div>
  <h1
    className="sc-gsnTZi jpFjlW"
    data-testid="mission-headline-test-id"
  >
    Our Mission & Vision
  </h1>
  <div>
    ────────
  </div>
  <p
    className="sc-eCYdqJ knwQMc"
  >
    Beam Dental was founded on the idea that everyone deserves access to dental care, centering our mission around closing the gap of the 100 million Americans who don't. By blending innovative technology with traditional insurance policies, we've been able to bring incredible value to the employee benefits market—both disrupting insurance forever and delivering on our mission.
    <br />
    <br />
     Beam’s vision is to deliver a fundamentally unique approach to coverage by incorporating dental hygiene behavior into policy pricing at renewal.
  </p>
  <div
    className="sc-hKMtZM dUReVK"
  >
    <img
      alt="eye glasses"
      src="glasses.png"
    />
  </div>
</div>
`;
